src/glossary.md

```diff
@@ -123,6 +130,25 @@ This is not affected by applied type arguments. `struct Foo` is considered local
 `Vec<Foo>` is not. `LocalType<ForeignType>` is local. Type aliases do not
 affect locality.
 
+### Name
+
+A [*name*] is an [identifier] or [lifetime or loop label] that refers to an
+[entity](#entity). A *name binding* is when an entity declaration introduces
+an identifier or label associated with that entity. [Paths],
+identifiers, and labels are used to refer to an entity.
+
+### Name resolution
+
+[*Name resolution*] is the compile-time process of tying [paths],
+[identifiers], and [labels] to [entity](#entity) declarations.
+
+### Namespace
+
+A [*namespace*] is a set of [names](#name) where each name unambiguously
+refers to a specific [entity](#entity). Namespaces are organized in a
+hierarchy, where each level of the hierarchy has its own collection of named
+entities.
```

Comment:
Looks like this is what is called "scope" in the compiler (https://github.com/rust-lang/rust/blob/master/compiler/rustc_resolve/src/lib.rs#L102-L115)?
"Namespace" in the compiler is one of "type", "value" and "macro" (+ perhaps "lifetime" and "label", but they are separate).

Reply:
Ah, ok, it's a "scope", but the wording is confusing.

>set of [names](#name) where each name unambiguously
refers to a specific [entity](#entity)

This is plain indecipherable.

>Namespaces are organized in a hierarchy

And this should somehow highlight that the hierarchies exist *inside* the namespaces, and not between them.

Reply:
Yea, I'm struggling to define it clearly. Lexicography is not my strong suit. I took another stab at it.

I suspect this will receive updates and clarifications once I start working on the "scopes" chapter.